### ISSUE TITLE
Exception raised on saving document

### ISSUE THREAD

**Issue by MikeWilliams-UK** [Chem4Word/Version3-1#54]
**Describe the bug**
The exception below is sometimes occurring when a document is saved.

**To Reproduce**
Steps to reproduce the behaviour:
Unknown

**Additional context**
```
OnDocumentBeforeSave()
System.Runtime.InteropServices.COMException (0xE0041804): The operation cannot be performed on a stream that is about to be deleted.
   at Microsoft.Office.Core._CustomXMLPart.Delete()
   at Chem4Word.Helpers.CustomXmlPartHelper.RemoveOrphanedXmlParts(Document doc) in C:\Dev\vsts\chem4word\Version3-1\src\Chem4Word.V3\Helpers\CustomXMLPartHelper.cs:line 133
   at Chem4Word.Chem4WordV3.OnDocumentBeforeSave(Document doc, Boolean& saveAsUi, Boolean& cancel) in C:\Dev\vsts\chem4word\Version3-1\src\Chem4Word.V3\Chem4WordV3.cs:line 1867
```

**Comment by tomyan112** [comment]
Can't reproduce

**Comment by MikeWilliams-UK** [comment]
@tomyan112 no problem I can put defensive code to silently handle this as it looks like Word has maybe already removed the document from memory, thus we can't do anything.

The code being called is to clean up any orphaned CML structures saved inside the document. If they are still present when the document is saved a sussequent time they should be cleaned up then.

**Comment by MikeWilliams-UK** [comment]
"Sticking plaster" in place - will be deployed with next version.

**Comment by MikeWilliams-UK** [comment]
Fixed in V3.1.8 Beta 8
